librz/include/rz_util/rz_str.h

```diff
@@ -236,8 +236,22 @@ RZ_API char *rz_str_replace_thunked(char *str, char *clean, int *thunk, int clen
 RZ_API bool rz_str_glob(const char *str, const char *glob);
 RZ_API int rz_str_binstr2bin(const char *str, ut8 *out, int outlen);
 RZ_API char *rz_str_between(const char *str, const char *prefix, const char *suffix);
-RZ_API bool rz_str_startswith(const char *str, const char *needle);
-RZ_API bool rz_str_startswith_icase(const char *str, const char *needle);
+/**
+ * \brief Checks if a string starts with a specifc sequence of characters (case sensitive)
+ * \param str C-string to be scanned
+ * \param needle C-string containing the sequence of characters to match
+ * \return True if \p needle is found at the beginning of \p str and false otherwise
+ * \see rz_str_startswith_icase()
+ */
+RZ_API bool rz_str_startswith(RZ_NONNULL const char *str, RZ_NONNULL const char *needle);
+/**
+ * \brief Checks if a string starts with a specifc sequence of characters (case insensitive)
+ * \param str C-string to be scanned
+ * \param needle C-string containing the sequence of characters to match
+ * \return True if \p needle is found at the beginning of \p str and false otherwise
+ * \see rz_str_startswith()
+ */
```

Comment:
same

Reply:
Okay. I've noticed some doxygen comments were placed in the headers for some functions and in the c files for other functions, I also didn't saw a recommendation regarding this in the [coding style](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#documentation). Is there any specific rationale for placing them in the source file btw?

Reply:
we have decided to place them in the C files. only structs in the headers contains the doxygen comments.

Reply:
I see, I meant I was interested to know the upsides/pros that lead the team to choose this particular style. Mostly of curiosity. Anyhow, will move them shortly.

Reply:
Also @wargio @ret2libc let me know if you think it's worth adding a line to the developer guide regarding the doxygen comments location

Reply:
IIRC the reason was that having the doxygen doc in the headers would make them too hard to navigate with all the text. At this point I think that being consistent is more important than anything else.

Reply:
I see